### ISSUE TITLE
Act-Eclipselink tests failure on tfb-status

### ISSUE THREAD

**Issue by greenlaw110** [TechEmpower/FrameworkBenchmarks#4874]
### OS (Please include kernel version)

TFB CI environment

### Expected Behavior

Test pass

### Actual Behavior

Test failure caused by

```
Exception in thread "main" java.lang.SecurityException: 
  class "javax.persistence.spi.PersistenceProviderResolverHolder$DefaultPersistenceProviderResolver$LoaderReference"'s 
  signer information does not match signer information of other classes in the same package
```

### Steps to reproduce behavior

TFB CI tests

### Other details and logs

https://tfb-status.techempower.com/unzip/results.2019-06-22-07-59-08-537.zip/results/20190618163411/act-eclipselink-mysql/run/act-eclipselink-mysql.log

What confuses me is 

* The travis CI doesn't complain the issue
* My local environment doesn't complain the issue (on Linuxmint 19.1, OpenJDK 8 64):
![image](https://user-images.githubusercontent.com/216930/59969785-a9093a80-9599-11e9-818f-d57c84f6efc8.png)

Can anyone help me identify what's wrong with the CI environment that triggered the issue?

**Comment by michaelhixson** [comment]
I can reproduce this locally with `tfb --mode verify --test act-eclipselink-mysql`.

When Googling the error message, I came across this StackOverflow question:  https://stackoverflow.com/questions/2877262/java-securityexception-signer-information-does-not-match

The answers indicate that it's likely two dependencies are providing the same set of classes, in particular the `javax.persistence.spi.PersistenceProviderResolverHolder` class from your stack trace.

I found a Maven plugin that can help find duplicate classes like this:
https://github.com/basepom/duplicate-finder-maven-plugin

I ran it like this:
```
mvn org.basepom.maven:duplicate-finder-maven-plugin:check -P eclipselink_mysql
```

It reported a lot of conflicts between these two dependencies, including the class from your stack trace:
```
[WARNING] Found duplicate and different classes in [org.eclipse.persistence:jakarta.persistence:2.2.2, org.hibernate.javax.persistence:hibernate-jpa-2.1-api:1.0.0.Final]:
...
[WARNING]   javax.persistence.spi.PersistenceProviderResolverHolder
```

Then I ran `mvn dependency:tree -P eclipselink_mysql` to see where those two dependencies were coming from.  Here's the relevant portion of that, showing how those two conflicting dependencies are being pulled in:
```
[INFO] +- org.actframework:act-eclipselink:jar:1.5.8:compile
[INFO] |  +- org.eclipse.persistence:org.eclipse.persistence.jpa:jar:2.7.4:compile
[INFO] |  |  +- org.eclipse.persistence:jakarta.persistence:jar:2.2.2:compile
[INFO] \- org.actframework:act:jar:1.8.22:compile
[INFO]    \- org.hibernate.javax.persistence:hibernate-jpa-2.1-api:jar:1.0.0.Final:compile
```

I don't know what the *right* solution is, but if I exclude one of those two dependencies by adding this to `frameworks/Java/act/pom.xml`...

```xml
<dependencyManagement>
  <dependencies>
    <dependency>
      <groupId>org.actframework</groupId>
      <artifactId>act</artifactId>
      <exclusions>
        <exclusion>
          <groupId>org.hibernate.javax.persistence</groupId>
          <artifactId>hibernate-jpa-2.1-api</artifactId>
        </exclusion>
      </exclusions>
    </dependency>
  </dependencies>
</dependencyManagement>
```

... then `tfb --mode verify --test act-eclipselink-mysql` succeeds.

Looking at the jar files from maven central...
* https://repo.maven.apache.org/maven2/org/eclipse/persistence/jakarta.persistence/2.2.2/jakarta.persistence-2.2.2.jar
* https://repo.maven.apache.org/maven2/org/hibernate/javax/persistence/hibernate-jpa-2.1-api/1.0.0.Final/hibernate-jpa-2.1-api-1.0.0.Final.jar

...I notice that the first one contains these two files, which I think are signature files:
* `META-INF/ECLIPSE_.RSA`
* `META-INF/ECLIPSE_.SF`

The other jar does not contain files like that.  So I think only the first jar is signed.

The right solution is probably one of these two:
* Rather than excluding one of those dependencies after the fact, change the act framework to not pull one of them in in the first place.  I'm not sure which dependency should be used.
* Somehow modify your `maven-assembly-plugin` configuration to exclude the signature files.

For the latter solution, in case it helps, I've noticed that most `maven-shade-plugin` examples include a bit like this:
```xml
<plugin>
  <groupId>org.apache.maven.plugins</groupId>
  <artifactId>maven-shade-plugin</artifactId>
  <configuration>
    <createDependencyReducedPom>false</createDependencyReducedPom>
    <filters>
      <filter>
        <artifact>*:*</artifact>
        <excludes>
          <exclude>META-INF/*.SF</exclude>
          <exclude>META-INF/*.DSA</exclude>
          <exclude>META-INF/*.RSA</exclude>
        </excludes>
      </filter>
    </filters>
  </configuration>
```

I never understood what that was doing, but now I think it is to avoid exactly the kind of error that you're seeing.  Since `maven-assembly-plugin` is used for similar purposes as `maven-shade-plugin`, maybe it also has a mechanism for excluding files like this.
